openmmtools/cache.py

```diff
@@ -459,7 +481,13 @@ def _generate_state_id(thermodynamic_state):
         """Return a unique key for the ThermodynamicState."""
         # We take advantage of the cached _standard_system_hash property
         # to generate a compatible hash for the thermodynamic state.
-        return str(thermodynamic_state._standard_system_hash)
+        return thermodynamic_state._standard_system_hash
+
+    @classmethod
+    def _generate_integrator_id(cls, integrator):
+        """Return a unique key for the given Integrator."""
+        standard_integrator = cls._standardize_integrator(integrator)
+        return openmm.XmlSerializer.serialize(standard_integrator).__hash__()
```

Comment:
I thought the `__hash__` here was non-deterministic. I see later on you use the `adler32` to hash out the integrator, but is it compatible here and/or should you be using the `adler32` hash here as well?